### DISCUSSION TITLE
If you were a billionaire ...

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2284] [General Discussion]
Picture this - you are a billionaire by whatever means. Money is no object to you anymore. You don't have to work (unless you want to), you have no particular obligations and you have plenty of spare time to do whatever you like. 

Now, you also happen to be a fervent MU* player/gm/etc and always dreamed of creating a MU* of your own. Maybe you are eccentric, but you just have no interest in graphical games.

So with essentially infinite resources ...

1. How would you go about the goal of making your dream MU*?
2. What would it be like?

**Comment by kovitikus** [comment]
I'd hire folk from the Evennia community first and foremost. I'd give them the choice to quit their day job and instead work with Evennia. They could choose to either work on the engine or work on my MUD. After that, if I still needed more developers, I'd hire from outside.

I'd also hire a really great writer that has the proper mindset for the type of dark and brutal world I want to create in my MUD. I'm not so great with writing lore, but then again I haven't spent enough time on it yet. I'm fairly adept at writing in general, but I haven't made the effort to hash out the details of any lore. Plus, my time would be better spent overseeing the aspects of the MUD, rather than focusing on hammering out every detail.

I'd hire an artist to help design the look and feel of the MUD and a custom client to go along with it. I'd use the extra development and artistic power to really make things crisp and snappy, with plenty of quality of life.

The MUD itself would have consistent content updates and "Seasons". PvP would be important to keep players from stagnating at a certain power level. Obviously the general concept of the MUD is the same as what I'm already making, with action combat, but on a much larger scale.

Something else that I'd spend my money on, that maybe people wouldn't consider, is advertising and marketing and "influencers" to stream my MUD and make videos about it. I believe that's the one big key aspect that graphical games have that MUDs don't, when it comes to enticing new and larger populations.

**Reply by ChrisLR** [reply]
Being paid to develop a mud.... epic dream

**Comment by Griatch** [comment]
@kovitikus The advertising/influencer aspect is certainly something that only money could buy, so-to-speak. Word of mouth is awesome, but if you have the money on a superbowl commercial or three, why not use it? :tophat:

**Reply by kovitikus** [reply]
Jeez. Could you imagine people's sheer confusion at a Super Bowl advertisement for MUDs? You'd have to make it really engaging and entertaining.

**Reply by Griatch** [reply]
Indeed.

**Comment by j3b4** [comment]
I'm not sure a billion would be enough.  I would want to acquire Google voice and stop it from being used to surveil my fellow citizens and instead dedicate all of its resources to developing an exquisitely responsive pure voice /audio mud client.   But why stop there. I'd like the audio mud to also become a pure voice IDE complete with conversational programming language where you develop code by discussing it with your (not google) Assistant.

**Reply by Griatch** [reply]
@j3b4 Why not - I didn't specify that it was a _single_ billion :smile: So you'd need a full-fledged AI for this then it sounds like (not only do the voice recognition but also be able to interpret and present the game world and 'interact')? 

Not quite a 'text' game anymore though in this case though ...

**Comment by Griatch** [comment]
Now that we are echoing properly to the support channels, maybe we can get some more input on this one!

**Comment by TehFamine** [comment]
I would likely make a game that you first have to find it out in the world, like a city or town. The game is loaded onto an arcade machine or something of that nature. It would be a MUD, then push the player from the machine to connecting on their own computer within another location. Then as they advance further, push them towards a 2d version of the game online, then a 3d version. Essentially, walking through the evolutions of gaming.

How you find the location of the arcade machine is from table-top content like pen&paper. Maybe a few books published on certain outlets that are hidden away for only a few souls to find and not knowing the book itself will start them on the path.

**Reply by Griatch** [reply]
So a mixture of real-world interaction and game. With that kind of resources I guess one doesn't need to care if only a few people find it either, one could even have multiple 'entrance points'. :grin:

**Reply by TehFamine** [reply]
Yup. I think it would be neat. I'm already rich so I can pay to support it until _someone_ finds it.

**Comment by Griatch** [comment]
In my case I'd fund a full-time team for Evennia-development itself - purely as a vanity project, with no aim of commercialization.
- A few good Python backend devs to help me refactor, cleanup and expand on the server itself. With this (and maybe a content-writer), one could also dramatically expand on the contribs and make a good Ainneve pretty quickly.
- A designer and a few frontend developers to (probably) rebuild the frontend app/webclient (likely as a single-page application) into something really spectacular while still being easy to expand on.
- A technical writer to help maintain/expand docs, write blogs etc.
- A hosting service for allowing people to set up, host and start to work on their Evennia game at the click of a button.
- Marketing team to help advertise the tech and get it noted by magazines etc.

**Reply by ChrisLR** [reply]
If that ever happens, I'm applying :D

**Comment by jaborsh** [comment]
You can make a billion dollars go a long away. Assuming I'm using every single penny to build the penultimate MUD experience, I'd spend a bit of time going through the present popular MUDs and determining what they do right and what they do wrong.

From spending a bit of time in the Evennia Discord Server and the r/MUD Discord Server, a lot of core complaints (the negatives that our money will be spent addressing) are:
  1. Core mechanics too simple/complex.
  2. Additional features and expanded mechanics are too simple/complex.
  3. Development is too slow.
  4. Administration is toxic.
  5. Balancing issues.

Core praises (the positives we'll be striving to emulate):
  1. Positive conflict environment.
  2. Unique classes (so, not another DnD copy/paste game).
  3. Unique story.
  4. The feeling of adrenaline during roleplay, exploring, etc.

All this said, the Billion Dollar Mud™ would look something like this:
  1. The introduction to the game would highlight the common necessities. Things like how to use 'say' or how to use 'emote', how to navigate the world, how to interact with NPCs, etc.
  2. The "high tech" features would expand upon one another as a player continually progresses. Doing this in a linear fashion will avoid the common pit-fall of a MUD flooding new players with brilliant ideas all at once. Often, it amplifies the havoc of the "learning phase". If things are segmented to the player instead of thrown hazardously at character generation, they're afforded ample time to appreciate what exactly the game has to offer.
  3. Development Team & HR Team. Separating the two ensures that people-friendly staff members are capable of taking the criticism and thanklessness that tends to be afforded by the common player and delivering it to the engineers without the latter group flipping out. A team dedicated to keeping everyone happy and heard eliminates problems that plague the MUD scene: sexual harassment, favoritism, nepotism, feeling overshadowed or individually unimportant, etc.
  4. One thing I'd like to see in a game - and ultimately I think this is one of the core things an upcoming MUD can do well regardless of its funding - is shifting the emphasis. Some games dedicate their focus to a unique and exciting combat system. Some games dedicate their focus to supporting a roleplay environment. Some games tell a story through their players. What I'm yet to see a MUD do is combine the aforementioned elements and encapsulate them into a rewarding quest/event system. I want to see a game that changes dramatically and in unforeseen ways.
       - A player stumbles upon a cave and accidentally antagonizes a lurking curse. All the descriptions throughout the game change, becoming dark and gloomy, the weather is dreary, the dead crawl out of the ground and attack.
       - A bolt of lightning strikes <insert important/magical monument here> and the city is razed to the ground. The playerbase has to go on a collective quest to revive <important monument> or rebuild the entire city by themselves.
       - A player finds a dark evil and together they conspire to murder <pivotal NPC>. The player succeeds and the game (and administration) support this newfound villain in a fashion that encourages further roleplay.

Now, you might say, "well, golly The-Archemist, you basically described a free-form roleplay game!" (like Arx; Arx is awesome). And yes, you could theoretically emulate all of these things in a game that is entirely story driven. The issue, however, is that in order for those things to happen, there has to be an individual leading the story. That's not necessarily a bad thing in and of itself, but no player can be present 24/7 to oversee the continual progression of a scene. When the Story Master is not present, individual roleplay or happenings that occur do not typically hold any significance beyond being a flavorful time-killer for people who make a hobby of writing.

If you've read this far or skipped to the end, I'll try and summarize my jumbled word vomit into a concise thought: my Billion Dollar Mud™ would empower the individual player to make world-impactful change without the necessity of a Story Master doing it on their behalf.

How this could be accomplished is up in the air. 

If you follow a model like Skyrim, it would come down to having a productive development team push out quest after quest and have the foresight to build the cause, effect, and conclusion in a way that doesn't require intervention.

If we're thinking futuristically, I'd like to think Artificial Intelligence has a lot of potential where MUDs are concerned: unique NPC interaction with players and spontaneously generated conversations depending on player input; combat diversity and gradual enemy improvement; "humanizing" the function of un-immersive robotic elements like RNG spawning, static NPC locations, NPCs never requiring sleep or having any other sort of routine; regenerating dungeons so that they maintain a new feeling. The list here is pretty endless.

**Reply by Griatch** [reply]
> If you'd read this far or skipped to the end, I'll try and summarize my jumbled word vomit into a concise thought: my Billion Dollar Mud™ would empower the individual player to make world-impactful change without the necessity of a Story Master doing it on their behalf. 

To be fair, if you were willing to spend one (or more?) Billions on this, you could expand this with a team of paid GMs in all time zones doing this around the clock ... ;) But I appreciate the technical beauty of allowing the user to self-serve too of course!

**Reply by jaborsh** [reply]
> To be fair, if you were willing to spend one (or more?) Billions on this, you could expand this with a team of paid GMs in all time zones doing this around the clock ... ;) But I appreciate the technical beauty of allowing the user to self-serve too of course!

Funny enough, I hadn't actually considered this. I was adhering to a mentality of `why spend six seconds doing something that I can automate in six hours?`.

That said, I realize that my response didn't really say what I've never seen a MUD do. I've never seen a MUD that prioritizes quests. There's plenty of roleplay MUDs, combat/grinding MUDs, ERP MUDs, but I can't think of one that encourages questing beyond radiant fetch quests.

**Reply by Griatch** [reply]
There are quest-based MUDs, but I guess anything that requires continuous content-creation will eventually run out of steam (well, unless you had a lot of money to pay content-creations of course!). Commercial MMORPGs spend a _lot_ of resources keeping their content fresh since users will run through it so quickly.

**Reply by jaborsh** [reply]
You're 100% right. I don't doubt their existence, but I think they're so buried under other games that don't require that continuous content-creation because, for the common passion-driven builder, it's hard.

That's where I'd put the billion dollars, be it with an expansive team of content creators, with a cleverly designed AI, or a mixture of the two.
